pkg/slo-controller/noderesource/noderesource_controller.go

```diff
@@ -86,21 +85,13 @@ func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	nodeMetric := &slov1alpha1.NodeMetric{}
 	if err := r.Client.Get(context.TODO(), req.NamespacedName, nodeMetric); err != nil {
-		if errors.IsNotFound(err) {
-			// skip non-existing node metric and return no error to forget the request
-			klog.V(3).Infof("skip for nodeMetric %v not found", req.Name)
-			return ctrl.Result{}, nil
-		}
-		klog.Errorf("failed to get nodeMetric %v, error: %v", req.Name, err)
-		return ctrl.Result{Requeue: true}, err
-	}
-
-	if r.isDegradeNeeded(nodeMetric, node) {
-		klog.Warningf("node %v need degradation, reset BE resource", req.Name)
-		if err := r.resetNodeResource(node, "degrade node resource because of abnormal nodeMetric, reason: "+degradeByKoordController); err != nil {
+		if !errors.IsNotFound(err) {
+			klog.ErrorS(err, "failed to get nodeMetric", "node", req.Name)
 			return ctrl.Result{Requeue: true}, err
 		}
-		return ctrl.Result{}, nil
+		nodeMetric = &slov1alpha1.NodeMetric{}
```

Comment:
Do we need to reset NodeMetric here? 
Please correct me if I'm wrong. I thought `nodeMetric` would remain empty if there is an error when calling `Get()`.

Reply:
OK. The nodeMetric object here is got by `CacheReader`, so it should remain empty when an error is returned.